internal/controllers/machinedeployment/machinedeployment_controller_test.go

```diff
@@ -343,72 +425,6 @@ func TestMachineDeploymentReconciler(t *testing.T) {
 			return len(machineSets.Items)
 		}, timeout*3).Should(BeEquivalentTo(1))
 
-		//
-		// Update a MachineDeployment spec.Selector.Matchlabels spec.Template.Labels
-		// expect Reconcile to be called and a new MachineSet to appear
-		// expect old MachineSets with old labels to be deleted
-		//
-		oldLabels := deployment.Spec.Selector.MatchLabels
-
-		// Change labels and selector to a new set of labels which doesn't have any overlap with the previous labels.
-		newLabels := map[string]string{
-			"new-key":                  "new-value",
-			clusterv1.ClusterNameLabel: testCluster.Name,
-		}
-
-		t.Log("Updating MachineDeployment labels")
-		modifyFunc = func(d *clusterv1.MachineDeployment) {
-			d.Spec.Selector.MatchLabels = newLabels
-			d.Spec.Template.Labels = newLabels
-		}
-		g.Expect(updateMachineDeployment(ctx, env, deployment, modifyFunc)).To(Succeed())
-
-		t.Log("Verifying if a new MachineSet with updated labels are created")
-		g.Eventually(func() int {
-			listOpts := client.MatchingLabels(newLabels)
-			if err := env.List(ctx, machineSets, listOpts); err != nil {
-				return -1
-			}
-			return len(machineSets.Items)
-		}, timeout).Should(BeEquivalentTo(1))
-		newms := machineSets.Items[0]
-
-		t.Log("Verifying new MachineSet has desired number of replicas")
```

Comment:
Is this part still covered?

Reply:
Added this check back. It was dropped by mistake. Thanks for catching it :)